### ISSUE TITLE
Out of order language tabs

### ISSUE THREAD

**Issue by Gijsdeman** [PaloAltoNetworks/docusaurus-openapi-docs#546]
## Describe the bug

When using the language tabs are configured in a different order than shown in [customization](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/blob/4b9ac927c85b56054c8d93336b1b5e282ab309e7/demo/docs/customization/languagetabs.md), the tabs seem to no longer switch properly between languages anymore.

## Expected behavior

The language to update when switching between language tabs.

## Current behavior

Language gets stuck on a language and does not load the language of the currently selected tab.

## Steps to reproduce

Adding the following `languageTabs` to the `themeConfig` will cause the language to render as intended.

`
            languageTabs: [
                {
                    highlight: "bash",
                    language: "curl",
                    logoClass: "bash",
                },
                {
                    highlight: "python",
                    language: "python",
                    logoClass: "python",
                    variant: "requests",
                },
                {
                    highlight: "go",
                    language: "go",
                    logoClass: "go",
                },
                {
                    highlight: "javascript",
                    language: "nodejs",
                    logoClass: "nodejs",
                    variant: "axios",
                },
                {
                    highlight: "php",
                    language: "php",
                    logoClass: "php",
                }
            ],
`

However, when changing the position of PHP language (or any other language), causes the tabs to no longer function properly.

`
            languageTabs: [
                {
                    highlight: "bash",
                    language: "curl",
                    logoClass: "bash",
                },
                {
                    highlight: "php",
                    language: "php",
                    logoClass: "php",
                },
                {
                    highlight: "python",
                    language: "python",
                    logoClass: "python",
                    variant: "requests",
                },
                {
                    highlight: "go",
                    language: "go",
                    logoClass: "go",
                },
                {
                    highlight: "javascript",
                    language: "nodejs",
                    logoClass: "nodejs",
                    variant: "axios",
                }
            ],
`

## Screenshots

![image](https://user-images.githubusercontent.com/43375368/232001632-94d3598b-4fdd-47ea-88ce-ce227580f27f.png)

![image](https://user-images.githubusercontent.com/43375368/232001724-4d696b48-f0b7-4dfc-bd5e-b25c9593c645.png)

## Context

PHP or shell has preference within the wanted documentation, so I prefer to have it as the first two options available. This is not a huge issue in general, but would be nice to be able to do this.

## Your Environment

Version used:
- docusaurus-plugin-openapi-docs: 1.7.0
- mozilla firefox for ubuntu: 111.0.1

**Comment by sserrata** [comment]
Hi @Gijsdeman, thanks for reporting. 

The `variant` is currently a required property. We'll update the examples in the documentation.

**Comment by sserrata** [comment]
Actually, it seems that it's only "required" once the default order is changed. This smells more like a bug. We'll investigate a bit more and will keep you updated.

**Comment by CheyiLin** [comment]
Same here with Docusaurus `2.4.1` and plugin `2.0.0-beta.3`.

```js
languageTabs: [
  {
    highlight: "bash",
    language: "curl",
    logoClass: "bash",
    variant: "cURL",
  },
  {
    highlight: "go",
    language: "go",
    logoClass: "go",
    variant: "native",
  },
  {
    highlight: "javascript",
    language: "nodejs",
    logoClass: "nodejs",
    variant: "axios",
  },
],
```

With the `languageTabs` config above, click tab `go` or `nodejs` will crash:

![image](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/assets/2572321/2144f916-8d55-4e99-9f4b-8d300b63d264)

**Comment by pepopowitz** [comment]
I also just ran into this. Some potentially helpful information: 

## Potential source of the bug

The crash message includes the "Available values" of variants for the current `languageTab`. Those available variant values appear to align with the _order_ of the default `languageTabs`, as defined in [CodeSnippets/index](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/blob/main/packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx#L29-L131). In other words, whichever `languageTab` you define 3rd will be looking for one of the variants defined as the 3rd default, i.e. [Go's ](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/blob/main/packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx#L53C3-L63) `native` variant.

I see that there's some config merging going on in that component. I suspect that the merging is resulting in most properties taking on the customized sequence of `languageTabs`, but not updating the associated `variants` property. (Which would make sense, if we're not defining the `variants` property in our reordered configuration.)

## A workaround

On a whim, I tried defining the `variants` property on each of the reordered `languageTabs` elements in my configuration. It works! The page no longer crashes, and all languages/variants display properly. 

So I guess if the tabs are reordered, both the `variant` _and_ `variants` properties become required, for any tab that is in a different position than its default.

As an example, this configuration for [the previous comment](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/issues/546#issuecomment-1663625785) functions correctly for me, with Docusaurus v`2.4.1` and plugin v`2.0.0-beta.4`: 

```javascript
languageTabs: [
  {
    highlight: "bash",
    language: "curl",
    logoClass: "bash",
    variant: "cURL",
  },
  {
    highlight: "go",
    language: "go",
    logoClass: "go",
    variant: "native",
    variants: ["native"],
  },
  {
    highlight: "javascript",
    language: "nodejs",
    logoClass: "nodejs",
    variant: "axios",
    variants: ["native", "axios", "request", "unirest"],
  },
],

```

**Comment by derekaug** [comment]
> I also just ran into this. Some potentially helpful information:
> 
> ## Potential source of the bug
> The crash message includes the "Available values" of variants for the current `languageTab`. Those available variant values appear to align with the _order_ of the default `languageTabs`, as defined in [CodeSnippets/index](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/blob/main/packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx#L29-L131). In other words, whichever `languageTab` you define 3rd will be looking for one of the variants defined as the 3rd default, i.e. [Go's ](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/blob/main/packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx#L53C3-L63) `native` variant.
> 
> I see that there's some config merging going on in that component. I suspect that the merging is resulting in most properties taking on the customized sequence of `languageTabs`, but not updating the associated `variants` property. (Which would make sense, if we're not defining the `variants` property in our reordered configuration.)
> 
> ## A workaround
> On a whim, I tried defining the `variants` property on each of the reordered `languageTabs` elements in my configuration. It works! The page no longer crashes, and all languages/variants display properly.
> 
> So I guess if the tabs are reordered, both the `variant` _and_ `variants` properties become required, for any tab that is in a different position than its default.
> 
> As an example, this configuration for [the previous comment](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/issues/546#issuecomment-1663625785) functions correctly for me, with Docusaurus v`2.4.1` and plugin v`2.0.0-beta.4`:
> 
> ```js
> languageTabs: [
>   {
>     highlight: "bash",
>     language: "curl",
>     logoClass: "bash",
>     variant: "cURL",
>   },
>   {
>     highlight: "go",
>     language: "go",
>     logoClass: "go",
>     variant: "native",
>     variants: ["native"],
>   },
>   {
>     highlight: "javascript",
>     language: "nodejs",
>     logoClass: "nodejs",
>     variant: "axios",
>     variants: ["native", "axios", "request", "unirest"],
>   },
> ],
> ```

also noticed that if I want to remove a variant / reorder a variant - it doesn't appear to work. for instance, I just wanted to show `native` and `axios` under nodejs but it always uses the default variants. I suspect this is due to how `lodash.merge` merges the variants array here: https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/blob/v3.0.0/packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx#L179

**Comment by sserrata** [comment]
adding to roadmap with the goal of expanding support for more languages and allow flexible ordering/inclusion

**Comment by sserrata** [comment]
Addressed in #862
